src/parse.js

```diff
@@ -22,6 +24,19 @@ const getEmptyBlock = () => ({
   },
 });
 
+const isTypeScript = (opt) => {
+  if (opt.sourceType === TYPESCRIPT) return true;
+  if (opt.filename) return opt.filename.endsWith('.ts') || opt.filename.endsWith('.tsx');
+  return false;
+};
+
+const getBabelParsingOptions = (useTypeScript) => {
+  if (useTypeScript) {
+    return { ...BABEL_PARSING_OPTS, plugins: ['typescript'].concat(BABEL_PARSING_OPTS.plugins) };
+  }
+  return { ...BABEL_PARSING_OPTS, plugins: ['flow'].concat(BABEL_PARSING_OPTS.plugins) };
+};
+
```

Comment:
Can these two functions be simplified to a single `getBabelParsingOptions(sourceType)`?

Reply:
Cool, I moved the `sourceType` construct in upper level function, it looks clear now